### PR TITLE
Bump go version to 1.20

### PR DESCRIPTION
**Pull request by liranr23**

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -104,7 +104,7 @@ jobs:
         uses: actions/setup-go@v2
         with:
           # NOTE: Keep the version in sync with Go toolchain in WORKSPACE.
-          go-version: '1.19.3'
+          go-version: '1.20.8'
```
Comment (ahadas) [inline .github/workflows/pull-request.yml:107]: why not 1.20.10?

File: WORKSPACE [hunk 2]
```diff
@@ -2,19 +2,19 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive", "http_file"
 
 http_archive(
     name = "io_bazel_rules_go",
-    sha256 = "ae013bf35bd23234d1dea46b079f1e05ba74ac0321423830119d3e787ec73483",
+    sha256 = "278b7ff5a826f3dc10f04feaf0b70d48b68748ccd512d7f98bf442077f043fe3",
     urls = [
-        "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.36.0/rules_go-v0.36.0.zip",
-        "https://github.com/bazelbuild/rules_go/releases/download/v0.36.0/rules_go-v0.36.0.zip",
+        "https://mirror.bazel.build/github.com/bazelbuild/rules_go/releases/download/v0.41.0/rules_go-v0.41.0.zip",
```
Comment (ahadas) [inline WORKSPACE:7]: we can even bump to 0.42, no?

**Review by ahadas — APPROVED** [review]

**Review by nyoxi — APPROVED** [review]

**Comment by nyoxi** [comment]
I don't know if it is OK to patch our vendored dependencies, but to make SonarCloud happy a change in govmomi is needed, specifically in `vendor/github.com/vmware/govmomi/Dockerfile.govc`.

```COPY --chown=appuser --from=build /temporary-tmp-directory /tmp```
to something like:
```COPY --chown=appuser --chmod=644 --from=build /temporary-tmp-directory /tmp```

**Comment by ahadas** [comment]
> I don't know if it is OK to patch our vendored dependencies, but to make SonarCloud happy a change in govmomi is needed, specifically in `vendor/github.com/vmware/govmomi/Dockerfile.govc`.
> 
> `COPY --chown=appuser --from=build /temporary-tmp-directory /tmp` to something like: `COPY --chown=appuser --chmod=644 --from=build /temporary-tmp-directory /tmp`

I suspect that our [ci job would then fail](https://github.com/kubev2v/forklift/blob/main/.github/workflows/pull-request.yml#L113). That's ok, as the risk is low, we can ignore that

**Pull request merged**